src/memory/memory.ts

```diff
@@ -2,29 +2,54 @@ import { InconsistentMemory, SegmentOutOfBounds } from 'errors/memory';
 
 import { Relocatable } from 'primitives/relocatable';
 import { SegmentValue } from 'primitives/segmentValue';
+import { BuiltinRunner, NoBuiltin } from 'builtins/builtin';
+
+export class Segment {
```

Comment:
why not extending `Array<SegmentValue>` ? see also https://stackoverflow.com/questions/2449182/getter-setter-on-javascript-array to update the getter for given builtins

Reply:
As discussed in a call, segments are refactored in [Proxy ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) of `Array<SegmentValue>`, builtins having a dedicated handler modifying the `[[Set]]` and/or `[[Get]]` methods to enforce their properties